Loading data from publicly available resources.

Dataset for YELP are retrieved from: https://business.yelp.com/data/resources/open-dataset/

In [ ]:
import pandas as pd
import tarfile
import json

tar_path = "yelp_dataset.tar"

rows = []
with tarfile.open(tar_path, "r:*") as tar:
    f = tar.extractfile("yelp_academic_dataset_review.json")

    for i, line in enumerate(f):
        rows.append(json.loads(line))
        if i == 99999:   # take first 1000 reviews
            break

df = pd.DataFrame(rows)
print(df.head())


                review_id                 user_id             business_id  \
0  KU_O5udG6zpxOg-VcAEodg  mh_-eMZ6K5RLWhZyISBhwA  XQfwVwDr-v0ZS3_CbbE5Xw   
1  BiTunyQ73aT9WBnpR9DZGw  OyoGAe7OKpv6SyGZT5g77Q  7ATYjTIgM3jUlt4UM3IypQ   
2  saUsX_uimxRlCVr67Z4Jig  8g_iMtfSiwikVnbP2etR0A  YjUWPpI6HXG530lwP-fb2A   
3  AqPFMleE6RsU23_auESxiA  _7bHUi9Uuf5__HHc_Q8guQ  kxX2SOes4o-D3ZQBkiMRfA   
4  Sx8TMOWLNuJBWer-0pcmoA  bcjbaE6dDog4jkNY91ncLQ  e4Vwtrqf-wpJfwesgvdgxQ   

   stars  useful  funny  cool  \
0    3.0       0      0     0   
1    5.0       1      0     1   
2    3.0       0      0     0   
3    5.0       1      0     1   
4    4.0       1      0     1   

                                                text                 date  
0  If you decide to eat here, just be aware it is...  2018-07-07 22:09:11  
1  I've taken a lot of spin classes over the year...  2012-01-03 15:28:18  
2  Family diner. Had the buffet. Eclectic assortm...  2014-02-05 20:30:30  
3  Wow!  Yummy, different,  delici

In [ ]:
df

review  rating
0      If you decide to eat here, just be aware it is...     3.0
1      I've taken a lot of spin classes over the year...     5.0
2      Family diner. Had the buffet. Eclectic assortm...     3.0
3      Wow!  Yummy, different,  delicious.   Our favo...     5.0
4      Cute interior and owner (?) gave us tour of up...     4.0
...                                                  ...     ...
99995  Came here for lunch with a group. They were bu...     4.0
99996  The equipment is so old and so felty! I just u...     1.0
99997  This is one of my favorite Mexican restaurants...     4.0
99998  Came here for brunch - had an omlette ($19 + t...     2.0
99999  Came in for my 5-6 month prophy and saw Kara -...     5.0

[100000 rows x 2 columns]

The kindle reviews Data is retrived from: https://www.kaggle.com/datasets/bharadwaj6/kindle-reviews


In [ ]:
import pandas as pd

df_am = pd.read_csv("kindle_reviews.csv")

In [ ]:
df_am

review  rating
0       I enjoy vintage books and movies so I enjoyed ...       5
1       This book is a reissue of an old one; the auth...       4
2       This was a fairly interesting read.  It had ol...       4
3       I'd never read any of the Amy Brewster mysteri...       5
4       If you like period pieces - clothing, lingo, y...       4
...                                                   ...     ...
982614  Yasss hunny! This is a great read. That Dre is...       5
982615  I ENJOYED THIS BOOK FROM BEGINNING TO END NOW ...       5
982616  Great book! Cherika was a fool. She let that m...       5
982617  When I say this was an excellent book please b...       5
982618  This book was everything. I just hope Alexus w...       5

[982619 rows x 2 columns]

In [ ]:

df = df[["text", "stars"]].rename(columns={"text": "review", "stars": "rating"})
df_am = df_am[["reviewText", "overall"]].rename(columns={"reviewText": "review", "overall": "rating"})


Getting the data scrapped from walmart using serpAPi

In [ ]:
import pandas as pd

df_wl = pd.read_csv("walmart_reviews.csv")

In [ ]:
df_wl

review  rating
0     This computer is made in such a way that if yo...       4
1     We purchased this laptop for our daughter who ...       5
2     When searching to purchasing ANYTHING from any...       5
3     In general, this is a very good $500 computer ...       4
4     Always check shipping company before you order...       3
...                                                 ...     ...
1077  Bought this skateboard for my son's Birthday. ...       1
1078  I bought this as I was starting to learn how t...       1
1079  Bearings don't roll trucks don't turn and the ...       1
1080  I had one and all i can say it is the worst qu...       1
1081                                       garbage dude       1

[1082 rows x 2 columns]

## Combining all data available and saving it.

In [ ]:
df_final = pd.concat(
    [df[['review', 'rating']], df_am[['review', 'rating']],df_wl[['review', 'rating']] ],
    ignore_index=True
)
df_final

review  rating
0        If you decide to eat here, just be aware it is...     3.0
1        I've taken a lot of spin classes over the year...     5.0
2        Family diner. Had the buffet. Eclectic assortm...     3.0
3        Wow!  Yummy, different,  delicious.   Our favo...     5.0
4        Cute interior and owner (?) gave us tour of up...     4.0
...                                                    ...     ...
1083696  Bought this skateboard for my son's Birthday. ...     1.0
1083697  I bought this as I was starting to learn how t...     1.0
1083698  Bearings don't roll trucks don't turn and the ...     1.0
1083699  I had one and all i can say it is the worst qu...     1.0
1083700                                       garbage dude     1.0

[1083701 rows x 2 columns]

In [ ]:
df_final.to_csv("df_complete_download.csv", index=False)

# df_human = df_human.sample(n=205760, random_state=42)

The commplete data has around "1083701" records and saved under
"df_final.csv" file.

And Also sampled down only to 50000 records selected randomly for the analysis.

In [ ]:
df_final = df_final.sample(n=50000, random_state=42)
df_final.to_csv("df_final.csv", index=False)

In [ ]:
df_sample = df_final.sample(n=1000, random_state=42)
df_sample.to_csv("df_sample.csv", index=False)